src/helper/sync.js

```diff
@@ -3,51 +3,66 @@
  */
 export default {
     events(parent) {
+        const sync = this;
+
         return function (method, model, options) {
             let xhr = null;
 
-            const trigger = function (triggerStr, ...args) {
-                if (model.trigger) {
-                    if (triggerStr.match(/success/)) {
-                        model.trigger(triggerStr, args[0]);
-                    } else {
-                        model.trigger(triggerStr);
-                    }
-                }
-            };
-
             // Name of the flag to keep track of requests.
             const flag = `inSync${method.charAt(0).toUpperCase()}${method.slice(1)}`;
 
             model[flag] = true;
 
-            trigger(`before:${method}`);
+            sync.trigger(model, `before:${method}`);
 
             xhr = parent.call(this, method, model, options);
 
             // Trigger 'after' event. If xhr exists, then request is in progress.
             // Otherwise something failed and cleanup.
             if (xhr) {
-                xhr.done((data, textStatus, jqXhr) => {
-                    model[flag] = false;
-                    trigger(`after:${method}:success`, data, textStatus, jqXhr);
-                });
-
-                xhr.fail((jqXhr, textStatus, errorThrown) => {
-                    model[flag] = false;
-                    trigger(`after:${method}:error`, jqXhr, textStatus, errorThrown);
-                });
-
-                xhr.always(() => {
-                    model[flag] = false;
-                    trigger(`after:${method}`);
-                });
+                xhr.done(sync.done(model, method, flag));
+                xhr.fail(sync.fail(model, method, flag));
+                xhr.always(sync.always(model, method, flag));
             } else {
                 model[flag] = false;
-                trigger(`after:${method}`);
+                sync.trigger(model, `after:${method}`);
             }
 
             return xhr;
         };
     },
+    trigger(model, triggerStr, ...args) {
+        if (model.trigger) {
+            if (triggerStr.match(/success/)) {
+                model.trigger(triggerStr, args[0]);
+            } else {
+                model.trigger(triggerStr);
+            }
+        }
+    },
+    done(model, method, flag) {
+        const sync = this;
+
+        return function (data, textStatus, jqXhr) {
+            model[flag] = false;
+            sync.trigger(model, `after:${method}:success`, data, textStatus, jqXhr);
+        };
+    },
+    fail(model, method, flag) {
+        const sync = this;
+
+        return function (jqXhr, textStatus, errorThrown) {
+            model; method; flag; jqXhr; textStatus; errorThrown;
```

Comment:
I highlighted line 55, but it gives the lines above that as context ;). So my question was about line 55.